Library/Homebrew/utils/shared_audits.rb

```diff
@@ -12,10 +12,15 @@ module SharedAudits
   def self.eol_data(product, cycle)
     @eol_data ||= T.let({}, T.nilable(T::Hash[String, T.untyped]))
     @eol_data["#{product}/#{cycle}"] ||= begin
-      result = Utils::Curl.curl_output("--location", "https://endoflife.date/api/#{product}/#{cycle}.json")
-      json = JSON.parse(result.stdout) if result.status.success?
-      json = nil if json&.dig("message")&.include?("Product not found")
-      json
+      result = Utils::Curl.curl_output("--location", "https://endoflife.date/api/v1/products/#{product}/releases/#{cycle}")
+
+      if result.status.success?
+        begin
+          JSON.parse(result.stdout)
+        rescue JSON::ParserError
+          nil
+        end
+      end
```

Comment:
What about this?
```suggestion
      JSON.parse(result.stdout) if result.status.success?
    rescue JSON::ParserError
      nil
```

Reply:
That's technically possible (since this is already in a `begin` block) but I'm not sure that we need to be using a `begin` block for more than just the `JSON.parse` call. I refactored the method to reduce the `begin` block to only the `JSON.parse` call, allowing us to use `return unless result.status.success?` instead.